pkg/scheduler/plugins/elasticquota/core/scale_minquota_when_over_root_res.go

```diff
@@ -0,0 +1,165 @@
+/*
+Copyright 2022 The Koordinator Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package core
+
+import (
+	"sync"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	quotav1 "k8s.io/apiserver/pkg/quota/v1"
+	"k8s.io/klog/v2"
+)
+
+type ScaleMinQuotaWhenOverRootResManager struct {
```

Comment:
The child nodes under each node will be divided into two categories, one allows scaling min quota, and the other does not allow. When our total resources are insufficient, the min quota of child nodes will be proportionally reduced. In order to calculate the scaling of the min quota, we will count the sum of the min quota of the child nodes of each node in advance. The quota of the child nodes that allow scaling is stored in parentToEnableScaleSubsSumMinQuotaMap, and the quota of child nodes that do not allow scaling is stored in parentToDisableScaleSubsSumMinQuotaMap.

Please add the above comments here.